### DISCUSSION TITLE
A way to correlate/serialize separate tasks requests to a specific external worker

### DISCUSSION THREAD

**Discussion by tonygionfriddo** [camunda-community-hub/community#64] [Q&A]
I have a use case where I need to maintain serialization of jobs based on a unique identifier.

If I build a workflow process with camunda, is there a way for me to either hash or correlate these jobs so they're not distributed to multiple workers and introduce an out of sequence / parallel processing scenario?

My use case demands that these different jobs with a correlating unique identifier are processed in the order they are received.

Any suggestions on how I can achieve this with external client workers?

I was reading about message correlation but not sure if that's the right answer for my scenario.

Thank you for all the help!

**Comment by jonathanlukas** [comment]
Hello @tonygionfriddo ,

as a job instance is locked for a duration as soon as it is activated by a worker, you can let as many workers as you like execute a job.

If you need to have jobs executed in a sequence, just model your process so that the service tasks come after each other. As long as the first one is not finished, the second one will not be created.

I hope this helps

Jonathan
